File: DAY 6/solutions.java/349. Intersection of Two Arrays.java
import java.util.*;

class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {

        HashSet<Integer> set1 = new HashSet<>();

        for (int num : nums1) {
            set1.add(num);
        }

        HashSet<Integer> answer = new HashSet<>();

        for (int num : nums2) {
            if (set1.contains(num)) {
                answer.add(num);
            }
        }

        int[] result = new int[answer.size()];

        int i = 0;

        for (int num : answer) {
            result[i] = num;
            i++;
        }

        return result;
    }
}

File: DAY 8/solutions.py/349. Intersection of Two Arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:

        hashmap = {}

        for num in nums1:
            hashmap[num] = 1

        result = []

        for num in nums2:

            if num in hashmap:
                result.append(num)
                del hashmap[num]

        return result